requests_toolbelt/multipart/encoder.py

```diff
@@ -568,3 +577,51 @@ def len(self):
 
     def read(self, length=-1):
         return self.fd.read(length)
+
+
+class FileFromURLWrapper(object):
+    """File from URL wrapper.
+
+    The :class:`FileFromURLWrapper` object gives you the ability to stream file
+    from provided URL in chunks by :class:`MultipartEncoder`.
+    Provide a stateless solution for streaming file from one server to another.
+
+    .. code-block:: python
+
+        import requests
+        from requests_toolbelt import MultipartEncoder, FileFromURLWrapper
+
+        encoder = MultipartEncoder(fields={'file': FileFromURLWrapper(url)})
+        r = requests.post('https://httpbin.org/post', data=encoder,
+                          headers={'Content-Type': encoder.content_type})
+    """
+
+    def __init__(self, file_url):
```

Comment:
Let's accept an optional session. This means that someone can do:

```py
import requests
from requests_toolbelt.multipart import encoder

session = requests.Session()
streaming_encoder = encoder.MultipartEncoder(fields={
    'file': encoder.FileFromURLWrapper(url, session=session),
})
```

This will then change our `__init__` to look like this:

```
def __init__(self, file_url, session=None):
    self.session = session or requests.Session()
    # what you have below
```